front/src/applications/editor/tools/switchEdition/tool.tsx

```diff
@@ -51,6 +60,46 @@ const SwitchEditionTool: Tool<SwitchEditionState> = {
         },
       },
     ],
+    [
+      {
+        id: 'delete-switch',
+        icon: BsTrash,
+        labelTranslationKey: `Editor.tools.switch-edition.actions.delete-switch`,
+        // Show button only if we are editing
+        isDisabled({ state }) {
+          return !state.initialEntity.geometry;
+        },
+        onClick({
+          infraID,
+          openModal,
+          closeModal,
+          forceRender,
+          state,
+          setState,
+          switchTypes,
+          dispatch,
+          t,
+        }) {
+          openModal(
+            <ConfirmModal
+              title={t(`Editor.tools.switch-edition.actions.delete-switch`)}
+              onConfirm={async () => {
+                await dispatch<ReturnType<typeof save>>(
+                  // We have to put state.initialEntity in array because delete initially works with selection which can get multiple elements
+                  // The assertion is required because of the Partial<SwitchEntity> returned by getNewSwitch which doesnt fit with EditorEntity
```

Comment:
```suggestion
                  // The cast is required because of the Partial<SwitchEntity> returned by getNewSwitch which doesnt fit with EditorEntity
```

Reply:
done :)